tests/flytekit/integration/remote/test_remote.py

```diff
@@ -1147,3 +1147,14 @@ def test_execute_workflow_with_dataclass():
         image_config=ImageConfig.from_images(IMAGE),
     )
     assert out.outputs["o0"] == ""
+
+
+def test_register_wf_twice(register):
+    from tests.flytekit.integration.remote.workflows.basic.pickle_wf import pickle_wf
+
+    remote = FlyteRemote(Config.auto(config_file=CONFIG), PROJECT, DOMAIN)
```

Comment:
When I revert the change in `flytekit/tools/script_mode.py` and `flytekit/types/pickle/pickle.py`, this test still passes.

Reply:
Updated it. I use the subprocess to register the workflow since flytekit won't compile `pickle_wf` again if I use flyte remote to register.